### PR TITLE
Better deletions on backend

### LINKED ISSUE #766: Create new endpoints for deleting projects/organizations
We need to have `delete` endpoints in project and organization controllers that would be used in SAVE.
These endpoints should simply update the status of the Project/Organization

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #766 

### What's done:
- [x] Implemented new endpoint for organization deletion
- [x] Implemented new endpoint for project deletion
- [x] Added tests on organization deletion
- [x] Added tests on project deletion

### CODE REVIEW

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt [hunk 1]
```diff
@@ -197,6 +192,27 @@ class ProjectController(
...
+     * @param project
+     * @param authentication
+     * @return response
+     */
+    @PostMapping("/delete")
+    fun deleteProject(@RequestBody project: Project, authentication: Authentication): Mono<StringResponse> =
+            projectService.findWithPermissionByNameAndOrganization(
+                authentication, project.name, project.organization, Permission.DELETE
+            )
+                .filter { projectPermissionEvaluator.hasPermission(authentication, project, Permission.DELETE) }
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:205]: Isn't it duplication of `findWithPermission...`?

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt [hunk 2]
```diff
@@ -114,6 +114,23 @@ internal class OrganizationController(
...
 
+    /**
+     * @param organizationName
+     * @param authentication
+     */
+    @DeleteMapping("/{organizationName}/delete")
+    @PreAuthorize("isAuthenticated()")
+    fun deleteOrganization(@PathVariable organizationName: String, authentication: Authentication): Mono<StringResponse> {
+        val role = lnkUserOrganizationService.getGlobalRoleOrOrganizationRole(authentication, organizationName)
+        val response = if (role.priority >= Role.OWNER.priority) {
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:125]: Can we use `organizationPermissionEvaluator` here, like you in ProjectController?

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt [hunk 3]
```diff
@@ -197,6 +192,27 @@ class ProjectController(
             ResponseEntity.ok(projectStatus.message)
         }
 
+    /**
+     * @param project
+     * @param authentication
+     * @return response
+     */
+    @PostMapping("/delete")
```
Comment (akuleshov7) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:200]: why not `DeleteMapping` here?

**Review by akuleshov7 — APPROVED** [review]

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt [hunk 4]
```diff
@@ -114,6 +114,23 @@ internal class OrganizationController(
...
+            organizationService.deleteOrganization(organizationName)
+            ResponseEntity.ok("Organization deleted")
+        } else {
+            ResponseEntity.status(HttpStatus.FORBIDDEN).build()
+        }
+        return Mono.just(response)
```
Comment (akuleshov7) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:131]: btw - please unify  the body of the function into a single Mono (functionally with `map`)

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt (hunk 3 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:200]: The point is that `WebTestServer` with `.delete()` method doesn't support `.body()` and the idea was to pass `Project` with a request, but now we decided to pass only `organizationName` and `projectName` instead.

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt [hunk 5]
```diff
@@ -114,6 +114,23 @@ internal class OrganizationController(
         return Mono.just(response)
     }
 
+    /**
+     * @param organizationName
+     * @param authentication
+     */
+    @DeleteMapping("/{organizationName}/delete")
+    @PreAuthorize("isAuthenticated()")
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:122]: No need, will delete it

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:205]: yep, seems like it is

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:125]: ok, will do it

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt (hunk 4 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:131]: ok, will do it

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt (hunk 3 above)
Comment (akuleshov7) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:200]: So? you decided to change parameters and will change Mapping?

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:200]: Yes, will use DELETE method instead of POST

**Review by petertrr — APPROVED** [review]

**Pull request merged**